src/web/utils/AppVisibilityUtils.ts

```diff
@@ -0,0 +1,98 @@
+/**
+* AppVisibilityUtils.ts
+*
+* Copyright (c) Microsoft Corporation. All rights reserved.
+* Licensed under the MIT license.
+*
+* Web-specific helpers for firing focus/activity related events
+*/
+import _ = require('./lodashMini');
+import SubscribableEvent from 'subscribableevent';
+
+const idleTimeInMs = 60 * 1000;
+
+export class AppVisibilityUtils {
+    private _isIdle = false;
+    private _timer: number|undefined;
+    
+    onFocusedEvent = new SubscribableEvent<() => void>();
+    onBlurredEvent = new SubscribableEvent<() => void>();
+    onAppForegroundedEvent = new SubscribableEvent<() => void>();
+    onAppBackgroundedEvent = new SubscribableEvent<() => void>();
+    onIdleEvent = new SubscribableEvent<() => void>();
+    onWakeUpEvent = new SubscribableEvent<() => void>();
+
+    constructor() {
+        window.addEventListener('focus', this._onFocus);
```

Comment:
I suspect that this is going to cause crashes in JS unit test frameworks that don't have a DOM since you're acting on the window and document object without checking if they exist

Based on existing comment in UserPresence.ts
`// Handle test environment where document is not defined.`

Reply:
Thanks Brent. Added it as a part of this pr https://github.com/Microsoft/reactxp/pull/710/files